pkg/common-controller/framework_test.go

```diff
@@ -1262,39 +1296,135 @@ func evaluatePVCFinalizerTests(ctrl *csiSnapshotCommonController, reactor *snaps
+		if funcName == "testRemovePVCFinalizer" {
+			for _, pvc := range reactor.claims {
+				if test.initialClaims[0].Name == pvc.Name {
+					if slice.ContainsString(test.initialClaims[0].ObjectMeta.Finalizers, utils.PVCFinalizer, nil) && !slice.ContainsString(pvc.ObjectMeta.Finalizers, utils.PVCFinalizer, nil) {
+						klog.V(4).Infof("test %q succeeded. PVCFinalizer is removed from PVC %s", test.name, pvc.Name)
+						bHasPVCFinalizer = false
+					}
+					break
+				}
+			}
+			if test.expectSuccess && bHasPVCFinalizer {
+				t.Errorf("Test %q: failed to remove finalizer from PVC %s", test.name, test.initialClaims[0].Name)
+			}
 		}
-	}
-	bHasPVCFinalizer = true
-	if funcName == "testRemovePVCFinalizer" {
-		for _, pvc := range reactor.claims {
-			if test.initialClaims[0].Name == pvc.Name {
-				if slice.ContainsString(test.initialClaims[0].ObjectMeta.Finalizers, utils.PVCFinalizer, nil) && !slice.ContainsString(pvc.ObjectMeta.Finalizers, utils.PVCFinalizer, nil) {
-					klog.V(4).Infof("test %q succeeded. PVCFinalizer is removed from PVC %s", test.name, pvc.Name)
-					bHasPVCFinalizer = false
+	} else {
+		if funcName == "testAddSnapshotFinalizer" {
+			for _, snapshot := range reactor.snapshots {
+				if test.initialSnapshots[0].Name == snapshot.Name {
+					if !slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) && slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) &&
+						!slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) && slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) {
+						klog.V(4).Infof("test %q succeeded. Finalizers are added to snapshot %s", test.name, snapshot.Name)
+						bHasSnapshotFinalizer = true
+					}
+					break
 				}
-				break
+			}
+			if test.expectSuccess && !bHasSnapshotFinalizer {
+				t.Errorf("Test %q: failed to add finalizer to Snapshot %s", test.name, test.initialSnapshots[0].Name)
 			}
 		}
-		if test.expectSuccess && bHasPVCFinalizer {
-			t.Errorf("Test %q: failed to remove finalizer from PVC %s", test.name, test.initialClaims[0].Name)
+		bHasSnapshotFinalizer = true
+		if funcName == "testRemoveSnapshotFinalizer" {
+			for _, snapshot := range reactor.snapshots {
+				if test.initialSnapshots[0].Name == snapshot.Name {
+					if slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) && !slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotBoundFinalizer, nil) &&
+						slice.ContainsString(test.initialSnapshots[0].ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) && !slice.ContainsString(snapshot.ObjectMeta.Finalizers, utils.VolumeSnapshotAsSourceFinalizer, nil) {
+						klog.V(4).Infof("test %q succeeded. SnapshotFinalizer is removed from PVC %s", test.name, snapshot.Name)
```

Comment:
s/SnapshotFinalizer is removed from PVC/SnapshotFinalizer is removed from snapshot